src/main/java/org/radarbase/management/config/SourceTypeLoader.java

```diff
@@ -134,21 +134,29 @@ public void saveSourceTypesFromCatalogServer(List<CatalogSourceType> catalogSour
             if (!sourceTypeRepository.findOneWithEagerRelationshipsByProducerAndModelAndVersion(
                     sourceType.getProducer(), sourceType.getModel(),
                     sourceType.getCatalogVersion()).isPresent()) {
-                // create new source-type
-                sourceTypeRepository.save(sourceType);
-
-                // create source-data for the new source-type
-                for (CatalogSourceData catalogSourceData : catalogSourceType.getData()) {
-                    SourceData sourceData = catalogSourceDataMapper
-                            .catalogSourceDataToSourceData(catalogSourceData);
-                    // sourceDataName should be unique
-                    // generated by combining sourceDataType and source-type configs
-                    sourceData.sourceDataName(sourceType.getProducer()
-                            + "_" + sourceType.getModel()
-                            + "_" + sourceType.getCatalogVersion()
-                            + "_" + sourceData.getSourceDataType());
-                    sourceData.sourceType(sourceType);
-                    sourceDataRepository.save(sourceData);
+                try {
+                    // create new source-type
+                    sourceTypeRepository.save(sourceType);
+
+                    // create source-data for the new source-type
+                    for (CatalogSourceData catalogSourceData : catalogSourceType.getData()) {
+                        try {
+                            SourceData sourceData = catalogSourceDataMapper
+                                    .catalogSourceDataToSourceData(catalogSourceData);
+                            // sourceDataName should be unique
+                            // generated by combining sourceDataType and source-type configs
+                            sourceData.sourceDataName(sourceType.getProducer()
+                                    + "_" + sourceType.getModel()
+                                    + "_" + sourceType.getCatalogVersion()
+                                    + "_" + sourceData.getSourceDataType());
+                            sourceData.sourceType(sourceType);
+                            sourceDataRepository.save(sourceData);
+                        } catch (RuntimeException ex) {
+                            log.error("Failed to import source data {}", catalogSourceData, ex);
+                        }
+                    }
+                } catch (RuntimeException ex) {
+                    log.error("Failed to import source type {}", sourceType, ex);
```

Comment:
Looks like `catch`es are only used for logging here. Not sure about the impact given the exceptions are swallowed.

Reply:
The exception swallowing is intentional. The previous behaviour was that exceptions were swallowed one level up, but then registration just stopped completely at the malformed source type.

Reply:
👍